src/Microsoft.Framework.Logging.Abstractions/LoggerExtensions.cs

```diff
@@ -581,17 +575,17 @@ public static void LogCritical([NotNull] this ILogger logger, int eventId, strin
             ILogValues state,
             Exception exception = null)
         {
-            logger.Log(logLevel, eventId, state, exception, _logValuesFormatter);
+            logger.Log(logLevel, eventId, state, exception, MessageFormatter);
         }
 
-        private static string LogValuesFormatter(ILogValues state, Exception exception)
+        private static string MessageFormatter(object state, Exception exception)
         {
             if (exception == null)
             {
                 return state?.ToString();
             }
 
-            return state?.ToString() + Environment.NewLine + exception;
+            return string.Format(CultureInfo.CurrentCulture, "{0}{1}{2}", state, Environment.NewLine, exception);
```

Comment:
Oh, state is a message? Well that's nice :smile: I really did mean log level + message. Imagine a call to `Logger.LogWarning("blah blah bad thing happened")` - that's super common. But not message? Maybe less common...